bindsnet/learning/__init__.py

```diff
@@ -119,10 +119,12 @@ def __init__(self, connection: AbstractConnection, nu: Optional[Union[float, Seq
                 'This learning rule is not supported for this Connection type.'
             )
 
-    def _connection_update(self, **kwargs) -> None:
+    def _connection_update(self, dt: float, **kwargs) -> None:
```

Comment:
The dt parameter can move to the _init_ function of the LearningRule class instead of adding it to all the _connection_update functions to prevent repetition.

Reply:
Actually, I like this approach better. It's like this for `Nodes` subclasses. That way, you can call `Network.run()` with different `dt` on different simulations.

Reply:
I don't understand. You mean network and connection update will have different dt?

Reply:
I mean, when you call `Network.run()`, you specify `dt` as an argument. This `dt` is used in `Nodes`, so it should also be used in `Connection`s and `LearningRule`s subclasses. This is somewhat contrived, but you could do something like:

```
network.run(inpts=inpts, dt=1.0, ...)  # Run with 1ms timestep
network.run(inpts=inpts, dt=0.1, ...)  # Run with 0.1ms timestep
```

You can use this to check for numerical instability.

If you passed in `dt` in `__init__` functions, you run the risk of accidentally assigning different `dt`s to different parts of the network, which would be a problem.

Reply:
If we have
`connection.dt = self.dt`
in the `network.add_connection function`, the update rule should have the same dt as the connection.

Reply:
Yeah, that would work. You could do that for `Nodes` and `LearningRule` objects, too. But then you'd want a function `Network.set_dt()` which propagated changes to child objects.

Reply:
I mean we already have this line in the network.add_connection function:
`connection.dt = self.dt`

And the following line in the network.add_layer function:
`layer.dt = self.dt`

So a simple thing would be to have : 
`connection.update_rule.dt = self.dt` below that. Thus preventing the need to pass a dt every time we want to update the connection and adding an additional parameter to each update_connection function. It would also automatically set the same dt for everything.

Reply:
Ah, I forgot about that. Sorry, I've been away from the codebase from a while. So let's have `self.update_rule.dt = connection.dt` in the `__init__` function of the `LearningRule` subclasses?

Reply:
Hmm, the problem is that the connection.dt is set when it is added to a network. So then the update_rule.dt will be none.

Reply:
So, what is the preferred approach here?

Reply:
Let's implement Devdhar's suggestion: use ‘self.connection.dt‘ in the update methods instead of passing it in to the learning rules' constructor.